src/WebJobs.Script/DependencyInjection/ExternalConfigurationStartupValidator.cs

```diff
@@ -12,11 +14,13 @@ internal class ExternalConfigurationStartupValidator
     {
         private readonly IConfiguration _config;
         private readonly IFunctionMetadataManager _metadataManager;
+        private readonly DefaultNameResolver _nameResolver;
 
         public ExternalConfigurationStartupValidator(IConfiguration config, IFunctionMetadataManager metadataManager)
         {
-            _config = config;
-            _metadataManager = metadataManager;
+            _config = config ?? throw new ArgumentNullException(nameof(config));
+            _metadataManager = metadataManager ?? throw new ArgumentNullException(nameof(metadataManager));
+            _nameResolver = new DefaultNameResolver(config);
```

Comment:
Ideally I'd use some NameResolverFactory here instead of hardcoding the type but I think this is okay for our scenario.